beacon/light/range.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package light
+
+// Range represents a (possibly zero-length) range of integers (sync periods).
+type Range struct {
+	First, Next uint64
```

Comment:
Ah, so as I undersand it after looking at it for a while, the `Range` is actually `[first, next[`, whereas start/end would be instead have to be `[start, end]`

Reply:
This is so far the best suggestion, accepted :) At one point it was "first" and "afterLast", that was probably the worst one.

Reply:
I just realized you meant [start, end] and not [start, end[. The reason I want the second value to point after the range it because I want to be able to handle zero length ranges without them being a special case most of the time.